scripts/build/builders/nxp.py

```diff
@@ -124,16 +136,19 @@ def __init__(self,
                  disable_ble: bool = False,
                  enable_thread: bool = False,
                  enable_wifi: bool = False,
-                 disable_ipv4: bool = False,
                  enable_shell: bool = False,
-                 enable_ota: bool = False):
+                 enable_ota: bool = False,
+                 enable_factory_data_build: bool = False,
+                 disable_pairing_autostart: bool = False,
+                 board_variant = None):
```

Comment:
what is the type of board_variant? Seems `Optional[str]`? However str seems a bit too loose ... could we make it an enum?

Reply:
Hello @andy31415 , that's correct here the `board_variant` type is "str", I updated the nxp.py file to specify the type. I have considered converting the `board_variant` into `enum` type, but currently keeping it as string seems to be more flexible. This approach allows to pass it easily to the Cmake option / GN arg which are expecting a string value. But I am open to suggestions, if there is a better way to manage the "board_variant" string while keeping compatibility with GN / cmake args :)

Reply:
Could you explain about the flexible part? Why should we not set it as a fixed enum? If we need to support more boards we need code changes in targets.py anyway, so it seems string just makes it vague and free form.

Reply:
About the flexible part, I was making reference to the fact that we are passing the string value of "board_variant" to a cmake option here : https://github.com/project-chip/connectedhomeip/pull/35448/commits/c1ddce018b9bd6255553b1535aa048dd73722c86#diff-16cf3d9dbea121fce0852e138b4ba2948b5f38c9d9b74699170a4cf75cc6551cR249 , because this information is needed for the build.
But I got your point, I have made a small rework to turn it into enum, please check my last commit :)